### PR TITLE
House keeping + Webpack 3

### PR DESCRIPTION
**Pull request by axe312ger**
Gives minor bundle size improvements plus potentual execution time improvements.

Also, cleans up the webpack config and separates the babel-preset-env config for browser and node

### CODE REVIEW

**Comment by axe312ger** [comment]
Most relevant changes are in https://github.com/contentful/contentful-management.js/pull/122/commits/ec04c85b19c05db41a5268c5ed863e47254694ae

**Comment by Khaledgarbaya** [comment]
@axe312ger can we avoid pushing changes related to linting next time, to make the PR more focus

File: webpack.config.js [hunk 1]
```diff
@@ -31,44 +30,56 @@ if (process.env.NODE_ENV === 'production') {
...
+          exclude: /(node_modules|bower_components|dist)/,
+          loader: 'babel-loader',
+          options: {
+            env: 'browser'
+          }
+        }
+      ]
     },
-    devtool: 'source-map',
+    devtool: PROD ? false : 'source-map',
```
Comment (Khaledgarbaya) [inline webpack.config.js]: I think this will cause a crash when the user try to build his app, because of `es6-promise` source maps

Comment (axe312ger) [inline webpack.config.js]: Why? This should stripe out source maps. IMHO a prod release should not contain any source maps, it is just clutter which extends the bundle size by 2

Comment (axe312ger) [inline webpack.config.js]: Also I had (accidentally) the es6-promise the the bundle with this setting and it was working fine? (At least bundeling and tests went green)

**Pull request merged**